docs/developerguide.rst

```diff
@@ -51,3 +51,422 @@ code directly editable within pods, you can use ``reana-cluster-dev.yaml``.
+and see your code changes on the new Pod which is created automatically.
+
+Note that you must manually mount the SSHFS mounts everytime you start working
+on REANA sources. It is also recommended that you unmount the sources when
+you stop working.
+
+To mount sources folder on remote host you would run the following command:
+
+.. code-block:: console
+
+   $ sshfs $USER@$FQDM:$REMOTE_PATH $LOCAL_PATH \
+   -o Compression=yes \
+   -o cache=yes \
+   -o kernel_cache \
+   -o follow_symlinks \
+   -o idmap=user \
+   -o no_remote_lock \
+   -o ServerAliveInterval=60 \
+   -o reconnect
+
+``$FQDN`` is URL to and ``$USER`` is your username at remote host running Minikube.
+``$REMOTE_PATH`` is the path on remote host where you will clone git
+repositories of REANA components and which will be later mounted to
+Minikube VM.
+``$LOCAL_PATH`` is the path on local machine which you want to map to remote host.
+
+To unmount you would run the following command:
+
+.. code-block:: console
+
+   fusermount -uzq $LOCAL_PATH
+
+where ``$LOCAL_PATH`` is the path on local machine where you have previously mounted
+sources of REANA components on remote host.
+
+**Use keyfile for authentication**
+
+In case you want to authenticate by a key file specify on with
+``-o IdentityFile=$KEYFILE_PATH`` option, where ``$KEYFILE_PATH`` is path to keyfile
+used to authenticate to remote host.
+
+**SSHFS and conenction encryption**
+
+SSHFS encrypts connections to remote host and depending on the encryption your
+local machine uses, file updates might be slow. To make file access faster
+one can use weaker encryption algorith for SSHFS connection using
+``-o Ciphers=arcfour`` option, but note that you must also enable this weak
+arcfour cipher on sshd config on remote host. Usually this is accomplished
+by adding set of allowed ciphers on sshd configuration file, which can usually
+be found in ``/etc/ssh/sshd_config``.
+
+Add following snippet to your sshd configuration file to allow use of arcfour
+cipher. Sshd evaluates values from left to right, so stronger ciphers will
+take preference and SSH client connecting to remote host will most likely
+have explicitly specify use of arcfour (as done with SSHFS).
+
+.. code-block:: console
+
+   # Defaults recommended by https://www.ssh.com/ssh/sshd_config/
+   # with addition of arcfour for fast SSHFS connections.
+   Ciphers aes256-gcm@openssh.com,aes128-gcm@openssh.com,aes256-ctr,aes192-ctr,aes128-ctr,chacha20-poly1305@openssh.com,arcfour
+
+**SSHFS and caching**
+
+In some setups you might benefit from explicitly configuring cache
+configuration values of SSHFS. Since file changes usually will happen only at
+local machine one can define quite long cache periods which prevent SSHFS
+to sync information of files that you haven't edited.
+SSHFS automatically invalidates cache on file that you edit.
+Following options have been observed to work OK, but no real performance
+measurements have been concluded.
```

Comment:
Would be cleaner if you move " but no real performance
+measurements have been concluded." after the option snippet